### DISCUSSION TITLE
讨论答案的正确性

### DISCUSSION THREAD

**Discussion by flymeqy** [MlgmXyysd/Xiaomi-BootLoader-Questionnaire#21] [General]
fastboot oem device-info：输出该设备的所有信息
关于此选项的正确性
我看到某安弄的模拟题里面是正确的，有冲突

**Comment by MlgmXyysd** [comment]
小米没加账号等级限制的时候，可以提交问卷来验证答案。在 2023.12.11 期，`fastboot oem device-info`命令的正确描述是`输出当前BL锁的状态`。
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/13d30ed7-978c-422c-a3ea-00ef41a3017a)
同时在早期题库（2023.11.08），对于描述`输出该设备的所有信息`，对应的命令为`FASTBOOT getvar all`。
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/c28a547a-f8f4-46a6-a91e-d24a815ba770)
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/b01a5607-414a-47f5-aa3a-7934a96bef72)
综上，`fastboot oem device-info：输出该设备的所有信息`应为错误描述。

**Reply by flymeqy** [reply]
4914260 同一个应用程序，versionCode变动时，versionName不可以保持不变

这个描述是错误的，因为互不影响，所以是可以保持不变的

**Reply by flymeqy** [reply]
4911384 手机如只能进入FASTBOOT，可能是引导程序损毁

这是18号的描述是＂不可能＂是引导程序损毁

**Reply by MlgmXyysd** [reply]
好的，我改一下，没看仔细hhhh

**Reply by MlgmXyysd** [reply]
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/52f173cb-108f-4a80-8002-9e2939bce00a)
18号抓包数据的描述是“可能是”

**Reply by flymeqy** [reply]
4923840 总线位宽是指总线上同时能够传输的数据位数，通常是指数据总线的根数

这句话的描述是正确的，刷题软件都有

**Reply by flymeqy** [reply]
采用CRC (Cyclic Redundancy Check) 差错校验时，首先需要将信息码字后面添加与生成多项式位数减一相同数量的0。对于您提供的例子，生成多项式G(X) = X^4 + X + 1的度数为4，所以要在信息码字后面添加4个0，变成10110000。

然后，使用二进制除法将带有额外0的信息码字除以生成多项式G(X)。在二进制除法中，我们不担心借位，只使用异或操作。计算过程为：

```
信息码字：    10110000  (添加了4个0)
生成多项式：   10011     (对应 X^4 + X + 1)
```

进行模2长除：

```
      10110
    ————————
10011 | 10110000
       - 10011
       ———————
        00101000
       -  10011
       ———————
         00011100
        -  10011
        ———————
          00001010
         -   10011  (无法再减，因为10011大于1010)
       ——————————
              1010  (余数就是CRC校验码)
```

余数即为所求的CRC校验码：0010。上述计算验证了您提供的CRC校验码0010是正确的。

这个每次计算结果可能不同，需要考虑计算过程中的正确性，这句话的描述大概率是正确的

**Reply by MlgmXyysd** [reply]
https://leventozturk.com/engineering/crc/
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/e2486a12-0ec6-40e9-b30a-75cba9743148)

**Reply by MlgmXyysd** [reply]
默认Initialise（00001）得出的结果也不是0010
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/3ee4c2ba-dff6-4d5f-be6c-21748d56846b)

**Reply by MlgmXyysd** [reply]
没有找到具体解释“总线位宽”的资料，以下为ChatGPT结论
![image](https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/17919858/f13fc7db-8916-4562-a6f0-c1d956905131)

**Reply by flymeqy** [reply]
昨天这题我全选了，有一个fastboot命令多选错了，得分92

**Reply by MlgmXyysd** [reply]
可以确定CRC是不对的，总线那题在获得更多的资料之前暂时不做修改

**Comment by flymeqy** [comment]
22号跟up主提供的参考答案几乎一模一样，才70多分，小米现在是根本没想给过啊。
答题视频连接https://www.123pan.com/s/eHmajv-8qBgh.html

**Reply by flymeqy** [reply]
fastboot那题为什么我选4个，因为选2个也是错的

**Reply by x364574152** [reply]
我也是今天刚答，和22号题目一样，答完72分，很迷

**Reply by MlgmXyysd** [reply]
考虑一下1501815的4924907选项，选项描述不是很清楚

**Reply by MlgmXyysd** [reply]
还有个4924918，改过来了现在
